From 0a352fc23a61c898ad0bbc3348304bec26d6b8de Mon Sep 17 00:00:00 2001
From: Rajat Soni <quic_rajson@quicinc.com>
Date: Wed, 16 Oct 2024 13:11:10 +0530
Subject: [PATCH] QSDK: CP: IW: Resolve compilation warnings

Some warnings are seen during iw compilation:

134917 scan.c:675:13: warning: 'print_ds' defined but not used [-Wunused-function]
134918   675 | static void print_ds(const uint8_t type, uint8_t len, const uint8_t *data,
134919       |             ^~~~~~~~
134920 scan.c:620:13: warning: 'print_rm_enabled_capabilities' defined but not used [-Wunused-function]
134921   620 | static void print_rm_enabled_capabilities(const uint8_t type, uint8_t len,
134922       |             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~

To resolve these warning we need to include compile these functions only when
IWFULL is enabled.

Fixes: I844b5a6304e ("iw: Adding new patch for build issue")

Patch-dependency: 501-compilation-fix.patch

Patch-work: None

Signed-off-by: Rajat Soni <quic_rajson@quicinc.com>
---
 scan.c | 15 +++++++--------
 1 file changed, 7 insertions(+), 8 deletions(-)

--- a/scan.c
+++ b/scan.c
@@ -615,7 +615,6 @@ static void print_supprates(const uint8_
 	}
 	printf("\n");
 }
-#endif
 
 static void print_rm_enabled_capabilities(const uint8_t type, uint8_t len,
 			    const uint8_t *data,
@@ -781,6 +780,7 @@ static void print_ap_channel_report(cons
 	}
 	printf("\n");
 }
+#endif
 
 static void print_cipher(const uint8_t *data)
 {
@@ -1140,7 +1140,6 @@ static const char* ntype_11u(uint8_t t)
 	default: return "Reserved";
 	}
 }
-#endif
 
 static const char* vgroup_11u(uint8_t t)
 {
@@ -1309,6 +1308,7 @@ static void print_tx_power_envelope(cons
 			printf("\t\t * %s: %i dBm\n", power_names[i], power_val);
 	}
 }
+#endif
 
 static const char *ht_secondary_offset[4] = {
 	"no secondary",
@@ -1512,8 +1512,6 @@ static void print_capabilities(const uin
 
 	printf("\n");
 }
-#endif
-
 
 static void print_tim(const uint8_t type, uint8_t len, const uint8_t *data,
 		      const struct print_ies_data *ie_buffer)
@@ -1531,6 +1529,7 @@ static void print_ibssatim(const uint8_t
 {
 	printf(" %d TUs\n", (data[1] << 8) + data[0]);
 }
+#endif
 
 static void print_vht_capa(const uint8_t type, uint8_t len, const uint8_t *data,
 			   const struct print_ies_data *ie_buffer)
@@ -1593,7 +1592,6 @@ static void print_supp_op_classes(const
 				break;
 		}
 }
-#endif
 
 static void print_measurement_pilot_tx(const uint8_t type, uint8_t len,
 				       const uint8_t *data,
@@ -1658,6 +1656,7 @@ static void print_obss_scan_params(const
 	printf("\t\t * OBSS Scan Activity Threshold: %d.%02d %%\n",
 		((data[13] << 8) | data[12]) / 100, ((data[13] << 8) | data[12]) % 100);
 }
+#endif
 
 static void print_secchan_offs(const uint8_t type, uint8_t len,
 			       const uint8_t *data,
@@ -1678,7 +1677,6 @@ static void print_bss_load(const uint8_t
 	printf("\t\t * channel utilisation: %d/255\n", data[2]);
 	printf("\t\t * available admission capacity: %d [*32us]\n", (data[4] << 8) | data[3]);
 }
-#endif
 
 static void print_mesh_conf(const uint8_t type, uint8_t len,
 			    const uint8_t *data,
@@ -1712,6 +1710,7 @@ static void print_mesh_conf(const uint8_
 	if (data[6] & 0x40)
 		printf("\t\t\t Mesh Power Save Level\n");
 }
+#endif
 
 struct ie_print {
 	const char *name;
@@ -1789,6 +1788,7 @@ static const struct ie_print ieprinters[
 #endif
 };
 
+#ifdef IW_FULL
 static void print_wifi_wpa(const uint8_t type, uint8_t len, const uint8_t *data,
 			   const struct print_ies_data *ie_buffer)
 {
@@ -2120,7 +2120,6 @@ static void print_wifi_wps(const uint8_t
 	}
 }
 
-#ifdef IW_FULL
 static const struct ie_print wifiprinters[] = {
 	[1] = { "WPA", print_wifi_wpa, 2, 255, BIT(PRINT_SCAN), },
 	[2] = { "WMM", print_wifi_wmm, 1, 255, BIT(PRINT_SCAN), },
@@ -2726,6 +2725,7 @@ COMMAND(scan, abort, "",
 	NL80211_CMD_ABORT_SCAN, 0, CIB_NETDEV, handle_scan_abort,
 	"Abort ongoing scan");
 
+#ifdef IW_FULL
 static int handle_start_sched_scan(struct nl80211_state *state,
 				   struct nl_msg *msg,
 				   int argc, char **argv, enum id_input id)
@@ -2743,7 +2743,6 @@ static int handle_stop_sched_scan(struct
 	return 0;
 }
 
-#ifdef IW_FULL
 COMMAND(scan, sched_start,
 	SCHED_SCAN_OPTIONS,
 	NL80211_CMD_START_SCHED_SCAN, 0, CIB_NETDEV, handle_start_sched_scan,
--- a/event.c
+++ b/event.c
@@ -103,7 +103,7 @@ static void print_frame(struct print_eve
 		printf(" %.02x", frame[i]);
 	printf("]");
 }
-#endif
+
 static void parse_cqm_event(struct nlattr **attrs)
 {
 	static struct nla_policy cqm_policy[NL80211_ATTR_CQM_MAX + 1] = {
@@ -297,6 +297,7 @@ static void parse_wowlan_wake_event(stru
 	if (tb[NL80211_WOWLAN_TRIG_WAKEUP_TCP_NOMORETOKENS])
 		printf("\t* TCP connection ran out of tokens\n");
 }
+#endif
 
 extern struct vendor_event *__start_vendor_event[];
 extern struct vendor_event *__stop_vendor_event;
@@ -304,6 +305,7 @@ extern struct vendor_event *__stop_vendo
 // Dummy to force the section to exist
 VENDOR_EVENT(0xffffffff, 0xffffffff, NULL);
 
+#ifdef IW_FULL
 static void parse_vendor_event(struct nlattr **attrs, bool dump)
 {
 	__u32 vendor_id, subcmd;
@@ -911,6 +913,7 @@ static void parse_assoc_comeback(struct
 	printf("assoc comeback bssid %s timeout %d\n",
 	       macbuf, timeout);
 }
+#endif
 
 static int print_event(struct nl_msg *msg, void *arg)
 {
--- a/interface.c
+++ b/interface.c
@@ -642,6 +642,7 @@ COMMAND(set, noack_map, "<[peer <addr>]
 	"Set the NoAck map for the TIDs. (0x0009 = BE, 0x0006 = BK, 0x0030 = VI, 0x00C0 = VO)");
 
 
+#ifdef IW_FULL
 static int handle_interface_wds_peer(struct nl80211_state *state,
 				     struct nl_msg *msg,
 				     int argc, char **argv,
@@ -669,7 +670,7 @@ static int handle_interface_wds_peer(str
  nla_put_failure:
 	return -ENOBUFS;
 }
-#ifdef IW_FULL
+
 COMMAND(set, peer, "<MAC address>",
 	NL80211_CMD_SET_WDS_PEER, 0, CIB_NETDEV, handle_interface_wds_peer,
 	"Set interface WDS peer.");
@@ -702,6 +703,7 @@ COMMAND(set, mcast_rate, "<rate in Mbps>
 	"Set the multicast bitrate.");
 
 
+#ifdef IW_FULL
 static int handle_chanfreq(struct nl80211_state *state, struct nl_msg *msg,
 			   bool chan, int argc, char **argv,
 			   enum id_input id)
@@ -763,7 +765,6 @@ static int handle_chan(struct nl80211_st
 	return handle_chanfreq(state, msg, true, argc, argv, id);
 }
 
-#ifdef IW_FULL
 SECTION(switch);
 COMMAND(switch, freq,
 	"<freq> [NOHT|HT20|HT40+|HT40-|5MHz|10MHz|80MHz|320MHz] [ru-puncturing-bitmap <bitmap>] [beacons <count>] [block-tx]\n"
--- a/station.c
+++ b/station.c
@@ -816,6 +816,7 @@ static const struct cmd *select_station_
 	return NULL;
 }
 
+#ifdef IW_FULL
 static int handle_station_set_plink(struct nl80211_state *state,
 			      struct nl_msg *msg,
 			      int argc, char **argv,
@@ -860,7 +861,7 @@ static int handle_station_set_plink(stru
  nla_put_failure:
 	return -ENOBUFS;
 }
-#ifdef IW_FULL
+
 COMMAND_ALIAS(station, set, "<MAC address> plink_action <open|block>",
 	NL80211_CMD_SET_STATION, 0, CIB_NETDEV, handle_station_set_plink,
 	"Set mesh peer link action for this station (peer).",
@@ -914,6 +915,7 @@ COMMAND_ALIAS(station, set, "<MAC addres
 	"Set an AP VLAN for this station.",
 	select_station_cmd, station_set_vlan);
 
+#ifdef IW_FULL
 static int handle_station_set_mesh_power_mode(struct nl80211_state *state,
 					      struct nl_msg *msg,
 					      int argc, char **argv,
@@ -960,7 +962,7 @@ static int handle_station_set_mesh_power
 nla_put_failure:
 	return -ENOBUFS;
 }
-#ifdef IW_FULL
+
 COMMAND_ALIAS(station, set, "<MAC address> mesh_power_mode "
 	"<active|light|deep>", NL80211_CMD_SET_STATION, 0, CIB_NETDEV,
 	handle_station_set_mesh_power_mode,
